comp/syntheticstestscheduler/impl/worker.go

```diff
@@ -0,0 +1,342 @@
+			return config.Config{}, fmt.Errorf("invalid TCP request type")
+		}
+		cfg.Protocol = payload.ProtocolTCP
+		cfg.DestHostname = req.Host
+		if req.Port != nil {
+			cfg.DestPort = uint16(*req.Port)
+		}
+		if req.MaxTTL != nil {
+			cfg.MaxTTL = uint8(*req.MaxTTL)
+		}
+		if req.Timeout != nil {
+			cfg.Timeout = time.Duration(*req.Timeout) * time.Second
+		}
+		cfg.TCPMethod = payload.TCPMethod(req.TCPMethod)
+		if req.SourceService != nil {
+			cfg.SourceService = *req.SourceService
+		}
+		if req.DestinationService != nil {
+			cfg.DestinationService = *req.DestinationService
+		}
+
+	case common.SubTypeICMP:
+		req, ok := c.Config.Request.(common.ICMPConfigRequest)
+		if !ok {
+			return config.Config{}, fmt.Errorf("invalid ICMP request type")
+		}
+		cfg.Protocol = payload.ProtocolICMP
+		cfg.DestHostname = req.Host
+		if req.MaxTTL != nil {
+			cfg.MaxTTL = uint8(*req.MaxTTL)
+		}
+		if req.Timeout != nil {
+			cfg.Timeout = time.Duration(*req.Timeout) * time.Second
+		}
+		if req.SourceService != nil {
+			cfg.SourceService = *req.SourceService
+		}
+		if req.DestinationService != nil {
+			cfg.DestinationService = *req.DestinationService
+		}
+
+	default:
+		return config.Config{}, fmt.Errorf("unsupported subtype: %s", c.Subtype)
+	}
+
+	return cfg, nil
+}
+
+// SyntheticsTestCtx is the unit of work consumed by workers.
+type SyntheticsTestCtx struct {
+	nextRun time.Time
+	cfg     common.SyntheticsTestConfig
+}
+
+// updateTestState updates lastRun and nextRun for a running test.
+func (s *SyntheticsTestScheduler) updateTestState(flushTime time.Time, rt *runningTestState) {
+	rt.lastRun = flushTime
+	rt.nextRun = flushTime.Add(time.Duration(rt.cfg.Interval) * time.Second)
+}
+
+// sendSyntheticsTestResult marshals the WorkerResult and forwards it via the epForwarder.
+func (s *SyntheticsTestScheduler) sendSyntheticsTestResult(w *WorkerResult) error {
+	res, err := s.networkPathToTestResult(w, s.configID)
+	if err != nil {
+		return err
+	}
+	payloadBytes, err := json.Marshal(res)
+	if err != nil {
+		return err
+	}
+	s.log.Debugf("synthetics network path test event: %s", string(payloadBytes))
+	m := message.NewMessage(payloadBytes, nil, "", 0)
+	return s.epForwarder.SendEventPlatformEventBlocking(m, eventplatform.EventTypeSynthetics)
+}
+
+// runTraceroute is the default traceroute execution using the traceroute package.
+func runTraceroute(cfg config.Config, telemetry telemetry.Component) (payload.NetworkPath, error) {
+	tr, err := traceroute.New(cfg, telemetry)
+	if err != nil {
+		return payload.NetworkPath{}, fmt.Errorf("new traceroute error: %s", err)
+	}
+	path, err := tr.Run(context.TODO())
+	if err != nil {
+		return payload.NetworkPath{}, fmt.Errorf("run traceroute error: %s", err)
+	}
+	return path, nil
+}
+
+// networkPathToTestResult converts a WorkerResult into the public TestResult structure.
+func (s *SyntheticsTestScheduler) networkPathToTestResult(w *WorkerResult, testConfigID string) (*common.TestResult, error) {
```

Comment:
Since the schema has changed a lot in datadog-traceroute, this code will need some update.

See example here: https://github.com/DataDog/datadog-agent/pull/40466